pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -48,24 +52,56 @@ func init() {
 // Add creates a new DatadogAgentDeployment Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+	reconciler, forwarders, err := newReconciler(mgr)
+	if err != nil {
+		return err
+	}
+	return add(mgr, reconciler, forwarders.Register)
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileDatadogAgentDeployment{client: mgr.GetClient(), scheme: mgr.GetScheme(), recorder: mgr.GetEventRecorderFor("DatadogAgentDeployment")}
+func newReconciler(mgr manager.Manager) (reconcile.Reconciler, metricForwardersManager, error) {
+	forwarders := datadog.NewForwardersManager(mgr.GetClient())
+	reconciler := &ReconcileDatadogAgentDeployment{
+		client:     mgr.GetClient(),
+		scheme:     mgr.GetScheme(),
+		recorder:   mgr.GetEventRecorderFor("DatadogAgentDeployment"),
+		forwarders: forwarders,
+	}
+	return reconciler, forwarders, mgr.Add(forwarders)
+}
+
+type metricForwardersManager interface {
+	Register(types.NamespacedName)
+	Unregister(types.NamespacedName)
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
+func add(mgr manager.Manager, r reconcile.Reconciler, registerFunc func(types.NamespacedName)) error {
```

Comment:
did you add this new parameters for unit-test purpose?

Reply:
no, `r` is an interface not a struct, I cannot access the `forwarders` attribute to use its `Register` method